src/puppetlabs/puppetdb/client.clj

```diff
@@ -42,7 +43,7 @@
    (let [body (json/generate-string payload)
          url (str (utils/base-url->str base-url)
                   (format "?command=%s&version=%s&certname=%s&producer-timestamp=%s"
-                          (str/replace command #" " "_") version certname (System/currentTimeMillis))
+                          (str/replace command #" " "_") version certname (c/from-long (System/currentTimeMillis)))
```

Comment:
Does the toString on a datetime instance automatically format it as the expected ISO8601?

Reply:
As long this part of puppetdb is also using jodatime, it does

https://www.joda.org/joda-time/apidocs/org/joda/time/base/AbstractDateTime.html#toString--